internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	secretmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret/scope"
+)
+
+func ResourceSecret() *schema.Resource {
+	return &schema.Resource{
+		Schema: secretSchema,
+	}
+}
+
+var secretSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	NamespaceNameKey: {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	OrgIDKey: {
+		Type: schema.TypeString,
+	},
+	scope.ScopeKey: scope.ScopeSchema,
+	specKey:        secretSpec,
+	ExportKey: {
+		Type:     schema.TypeBool,
+		Required: true,
+	},
+	statusKey: {
+		Type:     schema.TypeMap,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	},
+}
+
+// nolint: unused
+type dockerConfigJSON struct {
+	Auths map[string]map[string]interface{} `json:"auths,omitempty"`
+}
+
+var secretSpec = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the kubernetes secret",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			DockerConfigjsonKey: {
+				Type:        schema.TypeList,
+				Description: "SecretType definition - SECRET_TYPE_DOCKERCONFIGJSON, Kubernetes secrets type.",
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						UsernameKey: {
+							Type:        schema.TypeString,
+							Description: "SecretType definition - indicates the kubernetes username.",
+						},
+						PasswordKey: {
+							Type:        schema.TypeString,
+							Description: "SecretType definition - indicates the kubernetes password.",
+						},
+						ImageRegistryURLKey: {
+							Type:        schema.TypeString,
+							Description: "SecretType definition - indicates the kubernetes image registry url.",
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+// nolint: unused
```

Comment:
if this will be used in the next MR, then you can remove this lint and ensure the final feature branch passes